hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferContainer.java

```diff
@@ -141,8 +147,10 @@ private void initRingbufferStore(ClassLoader configClassLoader) {
         if (store.isEnabled()) {
             try {
                 final long storeSequence = store.getLargestSequence();
-                ringbuffer.setTailSequence(storeSequence);
-                ringbuffer.setHeadSequence(storeSequence + 1);
+                if (tailSequence() < storeSequence) {
```

Comment:
This was the bug with the ringbuffer store and the migrations. It goes like this - the ringbuffer would be migrated with some head and tail sequence (e.g. 0 and 100). After this, it would invoke the store and ask what's the highest sequence. The store returns 100 (since it stored all items) and then the ringbuffer unconditionally set the head and tail sequence to 100 and 101, thus emptying the in-memory state. 
This does not affect correctness but we unnecessarily lost in-memory items. The split brain tests cover this issue as the issue was detected because of the use of the ringbuffer store in the split brain tests, thanks to @Donnerbart